src/msg/async/AsyncConnection.cc

```diff
@@ -356,6 +356,8 @@ ssize_t AsyncConnection::_try_send(bool more)
     }
   }
 
+  last_active = ceph::coarse_mono_clock::now();
```

Comment:
I'm afraid the problem is more subtle as `last_active` is updated by the `write_callback_handler`. Would you mind elaborating a bit more why exactly this is not enough?

Reply:
Hi @rzarzynski , thank for your reply. 
Currently `last_active` is only updated by `write_callback_handler` in the data send direction and `process` in the receive direction, but `write_callback_handler` is only called when `AsyncConnection::writeCallback` is not null. `writeCallback` is assigned only when a call to `_try_send` does not send all data due to socket buffer size or other reasons. However, a connection between mgr and mgrc of a osd almost always can send a message from osd to mgr by one time call to `_try_send` without assigning `writeCallback`. And there is barely anything will send from mgr to osd, so the `last_active`  has no chance to be updated in this scenario.
This patch is trying to update `last_active` each time `_try_send` is called because when `_try_send` is called, it means data is being sent in this connection, so the active stamp of the connection should be updated.